packages/sdk/src/types/runtime/dialog.ts

```diff
@@ -0,0 +1,86 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { EventEmitter } from 'events';
+
+import { Context, GroupMask, Texture, User } from '.';
+
+/** The types of dialog boxes */
+export enum DialogType {
+	/** A dialog with text, an icon, and an OK button */
+	Message = 0,
+	/** A dialog with text, an icon, and OK and Cancel buttons */
+	Confirm,
+	/** A dialog with prompt text, a text input field, and OK and Cancel buttons */
+	Prompt
+}
+
+
+export enum DialogIcon {
```

Comment:
Make this a `type` and use string values:
```ts
export type DialogIcon
    = 'info'
    | 'warning'
    | 'error'
    | 'question'
    ;
```
In TypeScript, I want to avoid numeric enums wherever possible, and avoid having to prefix values with a type (e.g. DialogIcon.Info).

Reply:
I agree with avoiding numeric enums, but why no type prefixes? We do that with primitive types, and I think it aids in discoverability.

---
In reply to: [320499134](https://github.com/microsoft/mixed-reality-extension-sdk/pull/444#discussion_r320499134) [](ancestors = 320499134)

Reply:
The main benefit of this format over something prefixed is that it is directly representable in JSON, and this format seems to be where industry is going. For example, see the DOM definition file `C:\Users\<alias>\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\lib.dom.d.ts`. Toward the end of the file there are over 100 of these kinds of strongly-typed string unions defined.

Reply:
I think discoverability is good with string union types. Developers are used to them.